kpet/schema.py

```diff
@@ -158,43 +158,6 @@ def resolve(self, data):
         raise NotImplementedError()
 
 
-class Succession(Choice):
-    """
-    A schema describing a succession of accepted schema versions and the means
-    to inherit the legacy data - converter functions. Validates against one of
-    the schemas, recognizes as, and resolves to the last schema in the list.
-
-    Each uninterrupted sequence of supplied converter functions must accept
-    data validated by the preceding schema and return data validated by the
-    following schema.
-    """
-    def resolve(self, data):
-        self.validate(data)
-        # Last valid schema
-        last_valid_schema = None
-        # We find the first matching schema, then proceed converting and
-        # validating until we get to the last schema.
-        # For each schema/converter in the succession
-        for schema_or_converter in self.schemas_and_converters:
-            # If it's a schema
-            if isinstance(schema_or_converter, Node):
-                try:
-                    # Validate the data
-                    schema_or_converter.validate(data)
-                    last_valid_schema = schema_or_converter
-                except Invalid:
-                    # Cannot fail validation after a matching schema is found
-                    assert last_valid_schema is None
-            # Else it's a conversion function, and if we found valid schema
-            elif last_valid_schema:
-                # Convert the data for the next schema/converter
-                data = schema_or_converter(data)
-        # We should arrive at the last schema
-        assert last_valid_schema is self.schemas_and_converters[-1]
-        # Resolve the data with the last schema
-        return last_valid_schema.resolve(data)
-
-
```

Comment:
Please don't remove the Succession schema implementation, we'll keep using it :smile:!